core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -76,25 +87,62 @@ protected boolean requiresRelease() {
         return true;
     }
 
-    private ProbeResult fromSCMLink(ProbeContext context, String scm) {
+    private ProbeResult fromSCMLink(ProbeContext context, String scm, String pluginName) {
         Matcher matcher = GH_PATTERN.matcher(scm);
         if (!matcher.find()) {
             if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("{} is not respecting the SCM URL Template", scm);
+                LOGGER.debug("{} is not respecting the SCM URL Template.", scm);
             }
-            return ProbeResult.failure(key(), "SCM link doesn't match GitHub plugin repositories");
+            return ProbeResult.failure(key(), "SCM link doesn't match GitHub plugin repositories.");
         }
-
         try {
             context.getGitHub().getRepository(matcher.group("repo"));
-            return ProbeResult.success(key(), "The plugin SCM link is valid");
+            Optional<Path> file = searchPomFiles(context.getScmRepository(), pluginName);
+            String folderPath = file.isPresent() ? new File(file.toString()).getParentFile().getName() : "";
+            context.setScmFolderPath(folderPath);
+            return ProbeResult.success(key(), "The plugin SCM link is valid.");
         } catch (IOException ex) {
-            return ProbeResult.failure(key(), "The plugin SCM link is invalid");
+            return ProbeResult.failure(key(), "The plugin SCM link is invalid.");
         }
     }
 
     @Override
     public String[] getProbeResultRequirement() {
         return new String[]{UpdateCenterPluginPublicationProbe.KEY};
     }
+
+    /**
+     * Searches for Pom files in every directory available in the repository
+     *
+     * @param directory  in the scm
+     * @param pluginName the name of the plugin
+     * @return folderPath if it valid or return the scm itself
+     */
+    private Optional<Path> searchPomFiles(Path directory, String pluginName) {
+        Optional<Path> fileName = Optional.empty();
```

Comment:
Why do you need to initialize an `Optional` to empty?

Reply:
Because I cannot return the variable after `try` statement if the variable is not initialized. 

The other option is that I initialize it to `null`.

Reply:
I read the next comment. Which was related to this one and fixed it.

Reply:
I had this doubt of what to return after `try` statement. 

For now I am returning `null`.

Reply:
Do not return `null`, return `Optional.empty()`. You just don't need variables being assigned here.